crates/api/src/func.rs

```diff
@@ -149,15 +148,16 @@ macro_rules! getters {
     )*) => ($(
         $(#[$doc])*
         #[allow(non_snake_case)]
-        pub fn $name<'a, $($args,)* R>(&'a self)
-            -> anyhow::Result<impl Fn($($args,)*) -> Result<R, Trap> + 'a>
+        pub fn $name<$($args,)* R>(self)
```

Comment:
I think this would probably be best done by keeping `&self` as an argument, because that way if you want to test `Func` a few times you don't have to worry about cloning and getting the `Func` back in case of an error.

I think we can still return a `'static` closure here, it just needs to close over the `InstanceHandle` as well.

Reply:
Makes sense. I've now implemented this.